src/jdk.incubator.vector/share/classes/jdk/incubator/vector/ByteVector.java

```diff
@@ -445,6 +445,32 @@ static byte rotateRight(byte a, int n) {
         return (byte)(((((byte)a) & Byte.toUnsignedInt((byte)-1)) >>> (n & Byte.SIZE-1)) | ((((byte)a) & Byte.toUnsignedInt((byte)-1)) << (Byte.SIZE - (n & Byte.SIZE-1))));
     }
 
+    /* Implementation note: The implementation is based on Compress or Generalized Extract mentioned in
+     * Henry S. Warren, Jr's Hackers Delight, Addison Wesley, 2002.
+     */
+    static byte compressBits(byte a, byte bitmask) {
```

Comment:
According to the draft of 4th incubation(https://bugs.openjdk.java.net/browse/JDK-8280173), the `compress bit` should be similar to `vector mask compress`(CompressM). As per my understanding, the second argument `bitmask` should be an all true mask here. If so, it's also can be removed.

Reply:
I think i mistook the intended implementation when writing the JEP.

Since there is no scalar equivalent the operation is currently under specified, and in this case I think the expand operation also makes sense so `compress(expand(x, m), m) = x`

Reply:
Just talked with @sviswa7. I will update the JEP. Also, i will follow up on scalar integral implementations for compress/expand bits, where the `Byte`/`Short` implementations defer to that on `Integer`. Intrinsic scalar implementations can be implemented afterwards.

Reply:
Hi @PaulSandoz, do you think this implementation of compress bits is fine? I can plan to create a separate PR for expand bits operation. Do let me know. Thanks.

Reply:
> I think i mistook the intended implementation when writing the JEP.
> 
> Since there is no scalar equivalent the operation is currently under specified, and in this case I think the expand operation also makes sense so `compress(expand(x, m), m) = x`

If the true count of `m` is less than the significant bits of `x` , I think `x` will lost bits so that this equation doesn't work. E.g., m = 0b00000001, x = 0b11111111. It's 0b00000001 after expand(x, m) , and if compressed it back with the same bitmask, `m` will get 0b00000001.

Reply:
Yes, thanks, it only works for certain cases, i did not think it through carefully.